src/tfr_executive/src/clock_service.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * The clock service is a really simple node that exposes some utility services
+ * to simplify the operation of the executive system.
+ * 
+ * PARAMETERS 
+ * - ~driving_time: The amount of time we anticipate driving, to take in 
+ *                  seconds. (`float`, default 60)
+ * - ~dumping_time: The amount of time we anticipate dumping to take in 
+ *                  seconds. (`float`, default 30)
+ * - ~mission_time: The amount of time we allocate for the mission in total 
+ *                  seconds. (`float`, default: 600)
+ * SERVICES  
+ * 1. start_mission: starts the mission clock.
+ * 2. time_remaining: returns the amount of time remaining in the mission
+ * 3. digging_time: Gives the digging time which is equal to  
+ *                  (duration - start - driving_time - dumping_time) or 0 
+ *                  if negative duration. 
+ * */
+#include <ros/ros.h>
+#include <cmath>
+#include <tfr_msgs/EmptySrv.h>
+#include <tfr_msgs/DurationSrv.h>
+
+class ClockService
+{
+    public:
+        ClockService(ros::NodeHandle &n, ros::Duration& mission, 
+                ros::Duration& driving, ros::Duration& dumping):
+            start_mission{n.advertiseService("start_mission", &ClockService::startMission, this)},
+            time_remaining{n.advertiseService("time_remaining", &ClockService::timeRemaining, this)},
+            digging_time{n.advertiseService("digging_time", &ClockService::diggingTime , this)},
+            mission_start{},
+            mission_duration{mission},
+            driving_duration{driving},
+            dumping_duration{dumping}
+        {
+        }
+
+        ~ClockService() = default;
+        ClockService(const ClockService&) = delete;
+        ClockService& operator=(const ClockService&) = delete;
+        ClockService(ClockService&&) = delete;
+        ClockService& operator=(ClockService&&) = delete;
+
+
+    private: 
+        /*
+         * Starts the mission clock, takes in the empty service request.
+         * */
+        bool startMission(tfr_msgs::EmptySrv::Request &req,
+                tfr_msgs::EmptySrv::Response &res)
+        {
+            ROS_INFO("mission started");
+            mission_start = ros::Time::now();
+            return true;
+        }
+
+        /*
+         * Gives the remaining time in the mission, fills a duration request.
+         * */
+        bool timeRemaining(tfr_msgs::DurationSrv::Request &req,
+                tfr_msgs::DurationSrv::Response &res)
+        {
+            res.duration = mission_duration - (ros::Time::now() - mission_start);
```

Comment:
Below, in `diggingTime`, you do a check on `mission_start` to make sure it was initialized. Might as well do that same check here.

Reply:
Sounds good!